### ISSUE TITLE
build failed on Mojave with ansible playbook

### ISSUE THREAD

**Issue by wenglingmei** [m000/panda#2]
I tried to build panda in mojave and failed quickly for 
TASK [install pre-requisites] ******************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Updating Homebrew...\nError: No available formula with the name \"llvm@3.3\" \nError: No previously deleted formula found.\nError: No similarly named formulae found.\n==> Searching taps on GitHub...\nError: No formulae found in taps."}

I downloaded llvm33 binary and retry, and still failed
TASK [install pre-requisites] *****************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error: An exception occurred within a child process:\n  FormulaUnavailableError: No available formula with the name \"isl@0.18\""}

I know I can change the isl version in the playbook, but I was wondering if I missed any steps, since I saw the report that it should be compilable.

**Comment by m000** [comment]
That's weird. The formulas should have been copied from `resources/homebrew` to the homebrew formulas directory by the previous task. Can you add the following debug task before the `install pre-requisites` task in the playbook?

```
   - name: "formula debug"
     debug:
     msg: "{{homebrew_core_formulas}}"
```

What does this task print? Can you see `isl@0.11.rb` and `llvm@3.3.rb` among the files in this directory?

**Comment by wenglingmei** [comment]
> That's weird. The formulas should have been copied from `resources/homebrew` to the homebrew formulas directory by the previous task. Can you add the following debug task before the `install pre-requisites` task in the playbook?
> 
> ```
>    - name: "formula debug"
>      debug:
>      msg: "{{homebrew_core_formulas}}"
> ```
> 
> What does this task print? Can you see `isl@0.11.rb` and `llvm@3.3.rb` among the files in this directory?

Yes, they are there. The problem seems in the searching path of llvm@3.3. I worked around the issues with symbolic links. 

I tried several compilations. It always fails with an <cmath> error on a64. Is it belonging to the arm 64? Why macOS requires this?

**Comment by m000** [comment]
What kind of search path problems? The `llvm@3.3.rb` formula is copied to the main homebrew repo, so it should be automatically picked up. LLVM 3.3 should be installed on `$(brew --prefix)/opt/llvm@3.3/` and this location is passed to the configure script.

I'm not sure about the `a64` error. Could it be that it is `ia64`? It still wouldn't make sense because the architecture is not enabled at configure time.

I have tested this on the latest macOS (10.14 Mojave).

**Comment by wenglingmei** [comment]
In the configuration, it searched  /usr/local/Homebrew/opt instead of  /usr/local/opt. Is that the arm-softmmu?

**Comment by m000** [comment]
Ah... Can you share the output of `brew --config`? I believe in my setup some locations overlap, so it works, but it's not the general case.

ia64 is the Itanium architecture. It has been deprecated as a host, but I'm not sure if it's supported as guest by QEMU. Definitely not supported by PANDA though.

**Comment by wenglingmei** [comment]
HOMEBREW_VERSION: 2.1.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 37714b5ce1e4909d4d61de2af98343c7012f7cd9
Last commit: 5 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a6e5c1881a574c519d06afbf49d0f24c4bd25a20
Core tap last commit: 3 hours ago
HOMEBREW_PREFIX: /usr/local
CPU: quad-core 64-bit skylake
Homebrew Ruby: 2.3.7 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/bin/ruby
Clang: 11.0 build 1100
Git: 2.21.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_121
macOS: 10.14.6-x86_64
CLT: 10.3.0.0.1.1562985497
Xcode: 11.0
XQuartz: 2.7.11 => /opt/X11

I think you might be interested in the message error too. I reproduce it here:
TASK [configure PANDA] **********************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["./configure-osx.sh"], "delta": "0:00:05.517332", "end": "2019-09-29 17:42:56.559798", "msg": "non-zero return code", "rc": 1, "start": "2019-09-29 17:42:51.042466", "stderr": "./configure-osx.sh: line 8: /usr/local/Homebrew/opt/llvm@3.3/bin/llvm-config: No such file or directory", "stderr_lines": ["./configure-osx.sh: line 8: /usr/local/Homebrew/opt/llvm@3.3/bin/llvm-config: No such file or directory"], "stdout": "\nERROR: User requested feature curses\n       configure was not able to find it.\n       Install ncurses devel", "stdout_lines": ["", "ERROR: User requested feature curses", "       configure was not able to find it.", "       Install ncurses devel"]}

By the way, the clang/clang++ will be set to llvm3.3 and __thread is not supported, so I gave them the absolute path to the latest version clang/clang++.
I add cxxflag to suppress the errors on cmath, but I think there are still some settings missed. I got some weird error on objective-c.

**Comment by m000** [comment]
Funny. My config dumps more `HOMEBREW_*` locations
```
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CELLAR: /opt/homebrew/Cellar
HOMEBREW_REPOSITORY: /opt/homebrew
```

Can you also add the output of `brew --repository` and `brew --cellar` as well as the generated `configure-osx.sh`? Also, does homebrew bin directory come ahead of system bin directories in your `PATH`?

If you can make a good guess on what is off with the playbook, feel free to make a PR and I'll try it on my side.

**Comment by m000** [comment]
@wenglingmei I believe I've spotted the error. Formulas were removed by an implicit update of homebrew during prerequisite installation. Can you give it a try again?

**Comment by wenglingmei** [comment]
@m000 Sure, I will try it later today. 
---update---
I tried and the issue is still there. It still set to /usr/local/Homebrew/opt/llvm@3.3/bin/llvm-config in configure-osx.sh.
By the way, which version of SDK and command-line tool do you use? The version problem of toolkits in macOS is really a pain in the ass. So far, althougt the configuration gets passed, I still not able to build even only with the target of x86-64-softmmu.

I will test in a clean machine after a successful compilation.

**Comment by m000** [comment]
Hey, sorry for the long wait. I missed your last message because github doesn't send notifications when you update existing messages.

Here is my sdk configuration: 
```
$ xcrun --show-sdk-path
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
```
If you need to fix the path to `llvm-config` then this means that the `llvm_root` variable is not set right inside the playbook:
```
llvm_root: "{{homebrew_root}}/opt/llvm@3.3"
```

`{{homebrew_root}}` is filled with the output of the `brew --repo` command. Which apparently in your case prints `/usr/local/Homebrew`. Can you find which brew command prints `/usr/local` instead?

Also, what is the output of `brew shellenv` for your setup?
